### PR TITLE
Feature/eco bridge components refactor

### PR DESCRIPTION
**Pull request by denciu**
# Summary

- components that are part of `CurrencySearchInputPanel` have been restructured to keep files shorter and easier to maintain
- components that are nested in `CurrencySearchModal` have been refactored to use contextAPI. This way `CurrencySearchModal` (and also `CurrencySearchPanelInput`) can be reused on bridge
- `CurrencyLogo` refactor
- minor fixes in various files that were reported by `tsc` to have errors
- changed error message in `useEcoBridge`   

# To Test
To test on SWAP:
- [ ] user can add/remove custom list
- [ ] user can view information about lists
- [ ] disabling/enabling lists works as usual (tokens from selected list are visible on token list)

To test on BRIDGE:
- [ ] user is not able to add custom list
- [ ] user is not able to view information about lists
- [ ] disabling/enabling lists works as usual (tokens from selected list are visible on token list)

### CODE REVIEW

File: src/components/SearchModal/PairSearch/PairSearch.component.tsx [hunk 1]
```diff
@@ -79,16 +69,14 @@ export function PairSearch({ selectedPair, onPairSelect, onDismiss, isOpen, filt
...
-          if (
-            (
-              (filteredSortedPairs[0].token0.symbol || '') + (filteredSortedPairs[0].token1.symbol || '')
-            ).toLowerCase() === searchQuery.trim().toLowerCase() ||
-            filteredSortedPairs.length === 1
-          ) {
-            handlePairSelect(filteredSortedPairs[0])
-          }
+      if (e.key === 'Enter' && filteredSortedPairs.length > 0) {
+        if (
```
Comment (mix1o) [inline src/components/SearchModal/PairSearch/PairSearch.component.tsx:73]: Combine it to single `if`

File: src/services/EcoBridge/EcoBridge.hooks.ts [hunk 2]
```diff
@@ -121,7 +122,28 @@ export const useBridgeTokenInfo = (currency?: Currency, chainId?: ChainId): Wrap
 
 export const useBridgeActiveTokenMap = () => useSelector(selectBridgeActiveTokens)
 
-export const useBridgeSupportedLists = () => useSelector(selectSupportedLists)
+type WritableListsState = {
```
Comment (mix1o) [inline src/services/EcoBridge/EcoBridge.hooks.ts:125]: Move to `EcoBridge.types.ts`

File: src/components/CurrencyInputPanel/CurrencyInputPanel.component.tsx [hunk 3]
```diff
@@ -255,7 +145,12 @@ export default function CurrencyInputPanel({
           </InputRow>
           <FiatRow>
             <RowBetween>
-              {fiatValue && <FiatValueDetails fiatValue={fiatValue?.toFixed(2)} priceImpact={priceImpact} />}
+              {fiatValue && (
+                <FiatValueDetails
+                  fiatValue={fiatValue?.toFixed(2, { groupSeparator: ',' })}
```
Comment (berteotti) [inline src/components/CurrencyInputPanel/CurrencyInputPanel.component.tsx:150]: Nice!

File: src/components/CurrencyInputPanel/CurrencyInputPanel.container.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    debounceOnUserInput.cancel()
+  }, [debounceOnUserInput, disableCurrencySelect])
+
+  useBridgeInputValidation(!!disableCurrencySelect)
+
+  return (
+    <CurrencySearchModalContext.Provider value={currencySearchModalContext}>
+      <CurrencySearchContext.Provider value={currencySearchContext}>
+        <ManageListsContext.Provider value={manageListsContext}>
+          <ListRowContext.Provider value={listRowContext}>
```
Comment (berteotti) [inline src/components/CurrencyInputPanel/CurrencyInputPanel.container.tsx:84]: I see this is repeating. Should we define a `CurrencyInputContext` and use it in `CurrencyInputPanel` and `CurrencyInputPanelBridge`?

File: src/components/CurrencyLogo/index.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export { CurrencyLogo } from './CurrencyLogo.container'
+export { CurrencyLogoComponent } from './CurrencyLogo.component'
+export { CurrencyWrapperSource } from './CurrencyLogo.types'
+// eslint-disable-next-line
+export type { CurrencyLogoContainerProps } from './CurrencyLogo.types'
```
Comment (berteotti) [inline src/components/CurrencyLogo/index.ts:5]: I'm seeing this line too often. I think this fixes the error as seen here: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/consistent-type-exports.md#-correct-1

```suggestion
export { type CurrencyLogoContainerProps } from './CurrencyLogo.types'
```

Comment (denciu) [reply src/components/CurrencyLogo/index.ts:5]: I tried it and unfortunately I was not able to solve it this way even after disabling this rule as suggested in docs you mentioned. There is always some error with prettier/ts. I think we need to upgrade packages related to eslint and then come back to this. Meanwhile I found other solution which involves importing type with alias name and then reexport it again.

File: src/services/EcoBridge/EcoBridge.hooks.ts (hunk 2 above)
Comment (denciu) [reply src/services/EcoBridge/EcoBridge.hooks.ts]: Fixed

File: src/components/SearchModal/PairSearch/PairSearch.component.tsx (hunk 1 above)
Comment (denciu) [reply src/components/SearchModal/PairSearch/PairSearch.component.tsx]: Fixed

File: src/components/CurrencyInputPanel/CurrencyInputPanel.container.tsx (hunk 4 above)
Comment (denciu) [reply src/components/CurrencyInputPanel/CurrencyInputPanel.container.tsx]: Yep, good catch. Please look at `CurrencySearchModalProvider`

**Review by mix1o — APPROVED** [review]

**Comment by niemam29** [comment]
Not sure if it is or isn't on purpose but in the network switcher test nets are now not separated from other networks
cach2

**Comment by MilanVojnovic95** [comment]
Tested and looks ok to me.
the only thing I notice is strangely positioned indicated text and hyperlink, can we make it look better?

image

**Comment by denciu** [comment]
> Tested and looks ok to me. the only thing I notice is strangely positioned indicated text and hyperlink, can we make it look better?
> 
> image

I think it's the problem with the icon itself. If you look closely you can see that it's there and that's the reason for odd looking alignment. For different list on same branch:
image

**Review by karczuRF — APPROVED** [review]

**Pull request merged**